org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenExecutionContext.java

```diff
@@ -476,7 +485,7 @@ public IComponentLookup getComponentLookup() {
       throw new IllegalStateException();
     }
     if(projectFacade == null) {
-      return maven;
+      return containerLookup;
```

Comment:
Since we have a containerLookup at hand here, why isn't that always returned?
Is it because in case a projectFacade is present the project's classrealm is used as classloader?

Reply:
The passed lookup is only used for the case where we have not a project and otherwise depend on the project state, but probably can later be simplified... for now I just want to do it a bit step-by-step than one big bang :-)